src/components/services/Services.tsx

```diff
@@ -131,50 +132,42 @@ export function Services(props: Props) {
   // changes from off to on:
   useAsyncEffect(
     async () => {
-      let previousState = false
-      let isWarningShowing = false
-
-      const showBatterySaverWarning = async () => {
-        if (isWarningShowing) return
-
-        isWarningShowing = true
-        await Airship.show(bridge => <AlertDropdown bridge={bridge} message={lstrings.warning_battery_saver} warning persistent />)
-        isWarningShowing = false
-      }
-
-      const checkPowerSavingMode = async () => {
-        // This method is only available for Android
-        if (powerSavingOn != null) {
-          try {
-            previousState = await powerSavingOn()
-            if (previousState) {
-              // Show warning if power-saving mode is on initially
-              await showBatterySaverWarning()
-            }
-          } catch (error) {
-            showDevError(`Error checking power-saving mode: ${String(error)}`)
-          }
+      // This method is only available for Android
+      if (powerSavingOn == null) return
+
+      let airshipBridge: AirshipBridge<void> | undefined
+      const handlePowerSavingModeChanged = async (isPowerSavingModeOn: boolean) => {
+        if (isPowerSavingModeOn && airshipBridge == null) {
+          await Airship.show(bridge => {
+            airshipBridge = bridge // Capture the bridge here
+            return <AlertDropdown bridge={bridge} message={lstrings.warning_battery_saver} warning persistent />
+          }).then(() => {
+            airshipBridge = undefined
+          })
+        } else if (airshipBridge != null) {
+          // Dismiss the alert when power-saving mode turns off and there's an
```

Comment:
This will cause the alert to be dismissed if `isPowerSavingModeOn` and the modal is showing. While you shouldn't get this case, you are relying on distributed logic to ensure this doesn't break. Logic in the SDK in addition to logic in your code.

This should be

```
if (isPowerSavingModeOn && airshipBridge == null) {
   // showmodal
} else if (!isPowerSavingModeOn && airshipBridge != null) {
    // close modal
}
```

Reply:
Yeah I originally had it like this but assumed you'd prefer it less verbose since it shouldn't happen

Reply:
verbose isn't bad if it makes the logic crystal clear.

Reply:
Good to hear, that's my preference too